vendor_data.py

```diff
@@ -0,0 +1,156 @@
+        if (re.search(regex_mail,mail)):
+            self.VendorMail = mail
+        else:
+            raise ValueError("Please insert a valid email address.")      
+   
+    def to_excel(self):
+        data = {
+            "Target Language": [self.TargLang],
+            "Vendor": [self.VendorName],
+            "E-mail": [self.VendorMail],
+            "Word rate": [self.WordRate],
+            "CAT-tool": [self.CatTool]
+        }
+        filename = "_".join([str(self.ProjectName), str(self.SourceLang)])
+        df = pd.DataFrame(data)
+        if os.path.exists(f"{filename}.xlsx") != True:
+            df.to_excel(f"{filename}.xlsx", index=False, sheet_name="VendorData")
+        else:
+            with pd.ExcelWriter(f"{filename}.xlsx", mode="a", engine="openpyxl", if_sheet_exists="overlay") as writer:
+                df.to_excel(writer, sheet_name = "VendorData", startrow=writer.sheets["VendorData"].max_row, header = None, index=False)
+
+class PrefVendor(VendorData):
+    Preferred = True
+    Status = "Preferred"
+    """ Class Attribute:
+            Preffered (bool.) by default set to "True" indicating this is a Preferred Vendor, if "False" it is a back-up vendor."""
+    
+    def __init__(self, VendorName, TargLang, WordRate, VendorMail="", CatTool = "XTM"):
+        super().__init__(VendorName, TargLang, WordRate, VendorMail, CatTool)
+    
+    def to_excel(self):
+        data = {
+            "Target Language": [self.TargLang],
+            "Vendor": [self.VendorName],
+            "E-mail": [self.VendorMail],
+            "Word rate": [self.WordRate],
+            "CAT-tool": [self.CatTool]
+        }
+        filename = "_".join([str(self.ProjectName), str(self.SourceLang)])
+        df = pd.DataFrame(data)
+        if os.path.exists(f"{filename}.xlsx") != True:
+            df.to_excel(f"{filename}.xlsx", index=False, sheet_name="VendorData")
+        else:
+            with pd.ExcelWriter(f"{filename}.xlsx", mode="a", engine="openpyxl", if_sheet_exists="overlay") as writer:
+                df.to_excel(writer, sheet_name = "VendorData", startrow=writer.sheets["VendorData"].max_row, header = None, index=False)
+
+    def change_status(self):
+        self.Preferred = False
+        if self.Preferred == True:
+            self.Status = "Preferred"
+        else:
+            self.Status = "Back-Up"
+
+
+
+# don't forget to properly dcument them with docstrings!!
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser("docstring")
+    parser.add_argument("VendorName", type=str, help="Vendor's name")
+    parser.add_argument("TargLang", type=str, help="Language the vendor will be translating into")
+    parser.add_argument("--WordRate", type=float, help="The vendor's word rate in Euro")
+    parser.add_argument("--VendorMail", type=str, help="The vendor's email address")
+    parser.add_argument("--CatTool", type=str, help="The CAT-tool the vendor wants to use")
+    args = parser.parse_args()

```

Comment:
You can create a flag argument with the `action` attribute: https://docs.python.org/3/library/argparse.html#action
For example:

```python
parser.add_argument('-a', '--add', action='store_true')
```
Then `args.add` is True if the user calls `-a` or `--add` and False otherwise.